File: algoriths/java_code/insertion_sort.java
import java.util.*;
import java.io.*;

public class insertion_sort
{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        int size, i, j=0, temp=0;
        size = sc.nextInt();
        int a[] = new int[size];

        // reading the array elements
        for(i = 0; i < size; i++)
        {
            a[i] = sc.nextInt();
        }

        // insertion sort code
        // in-place sorting technique
        for(i = 1; i < size; i++)
        {
        	temp = a[i];
            for (j = i; j > 0; j--)
            {
            	if (a[j]<a[j-1]) {
            		a[j]=a[j-1];
            		a[j-1]=temp;
            	}

            }
        }
        
        // displaying the new sorted array
        for(i = 0; i < size; i++)
        {
            System.out.print(a[i]+", ");
        }

    }
}

File: algorithms/c&c++_code/insertion_sort.cpp
#include<iostream>
using namespace std;

int main(){
	int size,temp=0,i=0,j=0;
	cin>>size;	// read the size of array
	int a[size];
	
	// reading the elements
	for(i=0;i<size;i++){
		cin>>a[i];
	}
	
	// insertion sort
	for(i=1;i<size;i++){
		temp=a[i];
		for(j=i;j>0;j--){
			if(a[j]<a[j-1]){
				a[j]=a[j-1];
				a[j-1]=temp;
			}
		}
	}
	
	// displaying the elements in the array.
	for(i=0;i<size;i++){
		cout<<a[i]<<", ";
	}
	
	return 0;
}

